### ISSUE TITLE
more generator support for cmake dependencies

### ISSUE THREAD

**Issue by goulf-3m** [mesonbuild/meson#9305]
Here are code snippets in `./libtorch/share/cmake/Caffe2/Caffe2Targets.cmake`:

```
# Generated by CMake
...
# Generated CMake target import file.
...
set_target_properties(torch_library PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "\$<TARGET_PROPERTY:torch,INTERFACE_COMPILE_DEFINITIONS>"
  INTERFACE_COMPILE_OPTIONS "\$<TARGET_PROPERTY:torch,INTERFACE_COMPILE_OPTIONS>"
  INTERFACE_INCLUDE_DIRECTORIES "\$<TARGET_PROPERTY:torch,INTERFACE_INCLUDE_DIRECTORIES>"
  INTERFACE_LINK_LIBRARIES "-Wl,--no-as-needed,\"\$<TARGET_FILE:torch>\" -Wl,--as-needed;\$<TARGET_PROPERTY:torch,INTERFACE_LINK_LIBRARIES>"
  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "\$<TARGET_PROPERTY:torch,INTERFACE_SYSTEM_INCLUDE_DIRECTORIES>"
)
```

In meson, detecting libtorch (via cmake) fails.
As this file says it's generated, generators used are probably limited.

**Comment by mensinda** [comment]
Could you please provide the full `./libtorch/share/cmake/Caffe2` directory for implementing/debugging this issue?

**Comment by goulf-3m** [comment]
this is the whole cmake directory of libtorch:
[cmake.zip](https://github.com/mesonbuild/meson/files/7264750/cmake.zip)

The file I mentioned previously is generated by

```
  install(EXPORT Caffe2Targets DESTINATION share/cmake/Caffe2
      FILE Caffe2Targets.cmake
      COMPONENT dev)
```

**Comment by mensinda** [comment]
Should be fixed with #9743
